### PR TITLE
Add common snippets

### PR DESCRIPTION
**Pull request by hakonrossebo**
I've created some snippets that I find useful when programming in Elm. Do you think these can fit into vscode-elm? It's ok if you just want to use some of them. If not - should I put the snippets in a separate package?

### CODE REVIEW

File: snippets/elm.json [hunk 1]
```diff
@@ -91,6 +91,164 @@
...
+      ],
+      "description": "Outgoing port"
+    },
+    "Main Program": {
+      "prefix": "mainprogram",
+      "body": [
+        "main : Program Never",
+        "main =",
+        "  App.program",
+        "    { init = init",
```
Comment (hastebrot) [inline snippets/elm.json]: A snippet for `init : (Model, Cmd Msg)` could be nice.

File: snippets/elm.json [hunk 2]
```diff
@@ -91,6 +91,164 @@
...
+      "description": "Record definition"
+    },
+    "Update record": {
+      "prefix": "recordupdate",
+      "body": [
+        "{ ${recordName} | ${key1} = $1 }"
+      ],
+      "description": "Update record"
+    },
+    "Anonumous function": {
```
Comment (hastebrot) [inline snippets/elm.json]: s/Anonumous/Anonymous/

File: snippets/elm.json [hunk 3]
```diff
@@ -91,6 +91,164 @@
...
+      "prefix": "anonymous",
+      "body": [
+        "(\\ ${argument} -> ${argument} < 0)"
+      ],
+      "description": "Anonumous function"
```
Comment (hastebrot) [inline snippets/elm.json]: s/Anonumous/Anonymous/ again.

File: snippets/elm.json [hunk 4]
```diff
@@ -91,6 +91,164 @@
...
+      "prefix": "mainprogram",
+      "body": [
+        "main : Program Never",
+        "main =",
+        "  App.program",
+        "    { init = init",
+        "    , view = view",
+        "    , update = update",
+        "    , subscriptions = subscriptions",
+        "$1"
```
Comment (hastebrot) [inline snippets/elm.json]: Is the closing `}` missing here?

**Comment by hakonrossebo** [comment]
I've updated according to the comments. Also added a default model snippet.

**Comment by Krzysztof-Cieslak** [comment]
Thanks! :)

**Pull request merged**